Contribute/dotnet/dotnet-contribute.md

```diff
@@ -92,31 +110,23 @@ docs
             porting.vbproj
             porting-overview.vb
             Program.vb
+        /shared
+          /csharp ...
+          /fsharp ...
+          /vb ...
 ```
 
 > [!NOTE]
-> The language folders under snippets are not needed in the language guide area, where only one language is assumed.
+> The language folders under snippets are not needed in the language guide area, where only one language is assumed. For example, in the C# guide, it's assumed that all snippets are C#.
 
-The structure shown above includes one image, *portability_report.png*, and three code projects that include **code snippets** that are included in the *porting-overview.md* article. An accepted alternative structure contains one project per language that contains all snippets for all articles in that folder. This alternative has been used in the language reference areas because of very small snippets to demonstrate language syntax. It is discouraged in other areas.
+The structure shown above includes one image, *portability_report.png*, and three code projects that include **code snippets** that are included in the *porting-overview.md* article. 
 
-For historical reasons, many of the included snippets are stored under the */samples* folder in the *dotnet/docs* repository. If you're making major changes to an article, those snippets should be moved to the new structure. Do not move snippets for small changes.
+The *snippets/shared* folder is used for snippets that may span multiple articles within the same parent folder, such as the *porting* folder in the previous example. Only use the *shared* folder when you have a specific reason to do so, such as XAML code that's referenced by multiple articeles, yet can't compile in the *article-specific* folder.
 
-**Step 4:** Submit a Pull Request (PR) from your branch to the default branch.
+Media can also be shared across articles when those articles are in the same parent folder, such as the *porting* folder in the previous example. This *shared* folder should only be used when you have a specific reason to do so, such as a loading screen for an app or Visual Studio dialogs.
```

Comment:
>This *shared* folder should only be used when you have a specific reason to do so,

Does there have to be a reason other than the mere fact of use in multiple articles?

Reply:
Well, I was trying to communicate that we don't want to default to this behavior, it should be the exception.

```suggestion
Media can also be shared across articles when those articles are in the same parent folder, such as the *porting* folder in the previous example. This *shared* folder should be avoided if possible, and only used when it makes sense. For example, it may make sense to share a common loading screen for the app being demonstrated, or share Visual Studio dialogs that are reused across multiple articles.
```

Reply:
@tdykstra how is this?

Reply:
That makes the intent clear, thanks.